app/components/UserValidator/index.js

```diff
@@ -56,39 +60,73 @@ class Validator extends Component<Props, State> {
         }
       )
       .then(() => {
-        if (this.input) {
-          this.input.focus();
+        if (input) {
+          input.focus();
         }
       });
   };
 
-  render() {
-    return (
-      <div>
-        <div
-          className={cx(styles.overlay, {
-            [styles.shown]: this.state.showCompleted,
-          })}
-        >
-          <h3>
-            Tusen takk! Kos deg{' '}
-            <span role="img" aria-label="smile">
-              😀
-            </span>
-          </h3>
-          <i className="fa fa-check" />
-        </div>
-        <SearchPage
-          {...this.props}
-          placeholder="Skriv inn brukernavn eller navn"
-          handleSelect={this.handleSelect}
-          inputRef={(input) => {
-            this.input = input;
+  return (
+    <>
+      <div
+        className={cx(styles.overlay, {
+          [styles.shown]: completed,
+        })}
+      >
+        <h3>
+          Tusen takk! Kos deg{' '}
+          <span role="img" aria-label="smile">
+            😀
+          </span>
+        </h3>
+        <i className="fa fa-check" />
+      </div>
+      <Modal
+        contentClassName={styles.scannerModal}
+        show={showScanner}
+        onHide={() => setShowScanner(false)}
+      >
+        <h1>Scan ABA-ID</h1>
+        <QrReader
+          onResult={(res, error) => {
+            if (!processing && res) {
+              handleSelect({
+                username: res.getText(),
+                result: '',
+                color: '',
+                content: '',
+                icon: '',
+                label: '',
+                link: '',
+                path: '',
+                picture: '',
+                value: '',
+              });
```

Comment:
Instead of doing this, could you instead refactor `handleSelect` to only take the username? Only the username is used anyways, and with it nothing else would even work.

Reply:
Good point. Refactored it to only take the username. Also refactored SearchPage and Abacard to functional along the way.